Fixes #35411 - Add ouiaId prop to all supported components in host details

lfu opened the pull request:
#### What are the changes introduced in this pull request?
Add ouiaId prop to all supported components in host details

#### Considerations taken when implementing this change?

#### What are the testing steps for this pull request?
Follow instructions in comments in katello/webpack/global_test_setup.js
In katello dir, Run `npx jest webpack/components/extensions/HostDetails`

ianballou commented:
@lfu looks like the conflicts need resolution before I test.

ianballou commented:
Testing steps from @jeremylenz: "I'd check out PF's list of supported components here: https://www.patternfly.org/v4/developer-resources/open-ui-automation and then use the React dev tools to do some spot checks and make sure the ouiaId prop is added where it should be"

ianballou commented:
I'm seeing some missing ouiaIds:
1) Repository sets tab -> kebab Dropdown component that's on the right side of every row
2) Errata tab -> arrow Button component that's on the left side of every row
3) Traces tab -> Enable Traces Button component
4) Traces tab -> Click Enable Traces Button -> Enable Tracer Modal component
5) Traces tab -> Click Enable Traces Button -> Enable Tracer Button component and Cancel Button component
6) Errata tab -> DropdownToggle component on the top-right corner of the page for choosing pagination (with CaretDownIcon)

For all of these I'm seeing generated ouiaIds (data-ouia-component-id).  Jeremy mentioned something about generated components potentially not being able to get ouiaIds, perhaps some of those are mixed in here?

lfu commented:
> 1. Repository sets tab -> kebab Dropdown component that's on the right side of every row

ouiaId was added to DropdownItem in this PR but somehow it was not honored.
> 2. Errata tab -> arrow Button component that's on the left side of every row

This arrow is a [Td](https://github.com/Katello/katello/blob/eef975fa903a2ef9f079ab0365d47d2601977f74/webpack/components/extensions/HostDetails/Tabs/ErrataTab/ErrataTab.js#L527) element which does not accept ouiaId.
> 3. Traces tab -> Enable Traces Button component

Added.
> 4. Traces tab -> Click Enable Traces Button -> Enable Tracer Modal component

Added.
> 5. Traces tab -> Click Enable Traces Button -> Enable Tracer Button component and Cancel Button component

Added.
> 6. Errata tab -> DropdownToggle component on the top-right corner of the page for choosing pagination (with CaretDownIcon)

That DropdownToggle is defined by Pagination component and we don't have control over it. The Pagination has `ouiaId` set.

jeremylenz commented:
> ouiaId was added to DropdownItem in this PR but somehow it was not honored.

The actual kebab is a `<Td>` with `actions` prop. So if `<Td>` isn't supported, that's probably why.

lfu commented:
Test failure is unrelated.

ianballou commented:
I'm fixing the test failure here: https://github.com/Katello/katello/pull/10262

ianballou commented:
[test katello]

lfu commented:
[test katello]

ianballou commented:
I think I see some more:

1) Enable Tracer modal -> "via remote execution" Dropdown has a generated ouiaId
2) Content tab -> TabButton ouiaId is `undefined` (along with the other tabs including the Content subtabs) (`ForwardRef`)
3) Details -> Registration details -> "Registered by" Text component has a generated ouiaId
  -> Same with the other Text fields in the card
4) The repository sets page is broken so I cannot check it (seems on master too), but I saw that the "Return to last page" button on the error page did not have a ouiaId. (`ForwardRef`)
![image](https://user-images.githubusercontent.com/14796566/191063638-7031010c-5658-4671-a5e6-41b4a5c3e7df.png)
5) The Edit button in the top right doesn't have an ouiaId (I suppose that's probably in the Foreman code base -- so outside the scope of this PR?) (`ForwardRef`)

Note: some of these have `ForwardRef` in them as I've marked, so perhaps they can be ignored. I believe I remember @sjha4 mentioning that `ForwardRef` ones don't need an ouiaId? That might make sense because a lot of the missing ones have `ForwardRef`. I was confused though since some `ForwardRef` components do have ouiaIds.

lfu commented:
> 1. Enable Tracer modal -> "via remote execution" Dropdown has a generated ouiaId
![Screenshot from 2022-09-19 12-32-34](https://user-images.githubusercontent.com/2593270/191074855-5d0fa96d-39ff-4423-8488-6b140c6bc9cc.png)

> 3. Details -> Registration details -> "Registered by" Text component has a generated ouiaId

`Registered by` is a `DescriptionListTerm` which ends up as a `dt` element which does not require `ouiaId`

`ForwardRef` are components rendered by Patternfly so we can't add ouiaId's there.

ianballou commented:
> Enable Tracer modal -> "via remote execution" Dropdown has a generated ouiaId

This is what I'm seeing in the Component browser: 

![Screenshot from 2022-09-19 14-18-16](https://user-images.githubusercontent.com/14796566/191086562-27ee0ead-f248-4a38-b846-cbfd8acd47ee.png)


> Registered by is a DescriptionListTerm which ends up as a dt element which does not require ouiaId

I was looking at the Text component in this screenshot:

![image](https://user-images.githubusercontent.com/14796566/191086011-2c0d449b-1f60-464c-b3bd-08371e74d03c.png)

lfu commented on webpack/components/extensions/HostDetails/Tabs/TracesTab/EnableTracerModal.js:
```diff
@@ -43,7 +43,7 @@ const EnableTracerModal = ({ isOpen, setIsOpen, triggerJobStart }) => {
 
   const dropdownItems = dropdownOptions.map(text => (
-    <DropdownItem key={`option_${text}`} onClick={() => setSelectedOption(text)}>{text}</DropdownItem>
+    <DropdownItem key={`option_${text}`} ouiaId={`option_${text}`} onClick={() => setSelectedOption(text)}>{text}</DropdownItem>
```
@ianballou Do you see this change in your environment?

ianballou replied:
I do indeed

ianballou commented:
The one I see left is the Text field on the Details -> Registration details card. https://github.com/Katello/katello/blob/master/webpack/components/extensions/HostDetails/DetailsTabCards/RegistrationCard.js#L28

ianballou commented:
@lfu, rebasing should fix the error.

ianballou commented:
I found another Modal missing the ouiaId: https://github.com/Katello/katello/blob/master/webpack/components/extensions/HostDetails/Cards/SystemPurposeCard/SystemPurposeEditModal.js#L170

ianballou reviewed (approved):
I think we're all  good! Once https://github.com/Katello/katello/pull/10279 merges you can rebase and the test error should be fixed.

jeremylenz commented:
[test katello]

jeremylenz commented:
🍏

lfu commented:
Three more PRs for missing fields.
https://github.com/theforeman/foreman_remote_execution/pull/759
https://github.com/theforeman/foreman/pull/9448
https://github.com/Katello/katello/pull/10301
